x/swaprouter/keeper_test.go

```diff
@@ -0,0 +1,56 @@
+package swaprouter_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/osmosis-labs/osmosis/v13/app/apptesting"
+	"github.com/osmosis-labs/osmosis/v13/x/swaprouter/types"
+)
+
+type KeeperTestSuite struct {
+	apptesting.KeeperTestHelper
+}
+
+const testExpectedPoolId = 3
+
+var testPoolCreationFee = sdk.Coins{sdk.NewInt64Coin(sdk.DefaultBondDenom, 1000_000_000)}
+
+func TestKeeperTestSuite(t *testing.T) {
+
+	// TODO: re-enable this once swaprouter is fully merged.
+	t.SkipNow()
+
+	suite.Run(t, new(KeeperTestSuite))
+}
+
+func (suite *KeeperTestSuite) SetupTest() {
+	suite.Setup()
+}
+
+// createPoolFromType creates a basic pool of the given type for testing.
+func (suite *KeeperTestSuite) createPoolFromType(poolType types.PoolType) {
+	switch poolType {
+	case types.Balancer:
+		suite.PrepareBalancerPool()
+		return
+	case types.StableSwap:
+		// TODO
```

Comment:
Why cant we add SS to this in a state machine compatible way? Im only asking with the assumption that the next PR is the state break, so might as well get everything SM compatible in this PR.

Reply:
The tests for stableswap haven't been added yet. There is an issue to add that later